traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/SnapshotEventsProcessor.java

```diff
@@ -0,0 +1,376 @@
+		final Iterator<String> exServersIter = exCacheServers.fieldNames();
+		exServersIter.forEachRemaining(cId->{
+			if (!newServers.has(cId)) {
+				deleteCacheEvents.add(cId);
+			}
+		});
+	}
+
+	private List<DeliveryServiceReference> parseDsRefs(final JsonNode cacheJson) throws JsonUtilsException, ParseException{
+		final List<DeliveryServiceReference> dsRefs = new ArrayList<>();
+		if (cacheJson.has(ConfigHandler.deliveryServicesKey)) {
+			final JsonNode dsRefsJson = JsonUtils.getJsonNode(cacheJson, ConfigHandler.deliveryServicesKey);
+			final Iterator<String> dsrfs = dsRefsJson.fieldNames();
+			while (dsrfs.hasNext()) {
+				final String dsid = dsrfs.next();
+				final JsonNode dso = dsRefsJson.get(dsid);
+				DeliveryServiceReference newDsref = null;
+				if (dso.isArray() && dso.size() > 0) {
+					final JsonNode nameNode = dso.get(0);
+					final List<String> dsNames = parseDsAliases(dso);
+					newDsref = new DeliveryServiceReference(dsid, nameNode.asText(), dsNames);
+					dsRefs.add(newDsref);
+				} else {
+					dsRefs.add(new DeliveryServiceReference(dsid, dso.toString()));
+				}
+			}
+		}
+		return dsRefs;
+	}
+
+	private List<String> parseDsAliases(final JsonNode dso) {
+		final String tld = JsonUtils.optString(getExistingConfig(), "domain_name");
+		final List<String> dsNames = new ArrayList<>();
+
+		if (dso.isArray()) {
+			for (final JsonNode nameNode : dso) {
+				final String name = nameNode.asText();
+
+				if (name.endsWith(tld)) {
+					final String reName = name.replaceAll("^.*?\\.", "");
+
+					if (!dsNames.contains(reName)) {
+						dsNames.add(reName);
+					}
+				} else {
+					if (!dsNames.contains(name)) {
+						dsNames.add(name);
+					}
+				}
+			}
+		} else {
+			dsNames.add(dso.toString());
+		}
+	    return dsNames;
+	}
+
+	private void parseDeliveryServices(final JsonNode newSnapDb) throws
+			JsonUtilsException {
+		this.diffDeliveryServices(newSnapDb, null);
+	}
+
+	private void diffDeliveryServices(final JsonNode newSnapDb, final JsonNode existingDb) throws
+			JsonUtilsException {
+		JsonNode compDeliveryServices = null;
+		final JsonNode newDeliveryServices = JsonUtils.getJsonNode(newSnapDb, ConfigHandler.deliveryServicesKey);
+		final List<String> existingIds = new ArrayList<>();
+
+		// find delivery services that have been changed or deleted in the new snapshot
+		if (existingDb != null) {
+			compDeliveryServices = JsonUtils.getJsonNode(existingDb, ConfigHandler.deliveryServicesKey);
+
+			final Iterator<String> compIter = compDeliveryServices.fieldNames();
+			while (compIter.hasNext()) {
+				final String deliveryServiceId = compIter.next();
+				final JsonNode compDeliveryService = JsonUtils.getJsonNode(compDeliveryServices, deliveryServiceId);
+				existingIds.add(deliveryServiceId);
+
+				final JsonNode newService = newDeliveryServices.get(deliveryServiceId);
+
+				if (newService != null) {
+					if (isUpdated(newService, compDeliveryService)) {
```

Comment:
this `if` can be an `&&` in the one above it

Reply:
done